lib/common-types.ts

```diff
@@ -0,0 +1,175 @@
+  total: number;
+  skip: number;
+  limit: number;
+  items: Array<T>;
+}
+export type AssetCollection = ContentfulCollection<Asset>;
+export interface Entry<T> {
+  sys: Sys;
+  fields: T;
+}
+export interface EntryCollection<T> extends ContentfulCollection<Entry<T>> {
+  errors?: Array<any>;
+  includes?: any;
+}
+export interface ContentType {
+  sys: Sys;
+  name: string;
+  description: string;
+  displayField: string;
+  fields: Array<Field>;
+}
+export declare type ContentTypeCollection = ContentfulCollection<ContentType>;
+export interface Locale {
+  code: string;
+  name: string;
+  default: boolean;
+  fallbackCode: string | null;
+  sys: {
+    id: string;
+    type: 'Locale';
+    version: number;
+  };
+}
+export declare type LocaleCollection = ContentfulCollection<Locale>;
+export interface SyncCollection {
+  entries: Array<Entry<any>>;
+  assets: Array<Asset>;
+  deletedEntries: Array<Entry<any>>;
+  deletedAssets: Array<Asset>;
+  nextSyncToken: string;
+}
+export interface Sys {
+  type: string;
+  id: string;
+  createdAt: string;
+  updatedAt: string;
+  locale: string;
+  revision?: number;
+  space?: {
+    sys: SpaceLink;
+  };
+  contentType: {
+    sys: ContentTypeLink;
+  };
+}
+export interface SpaceLink {
+  type: 'Link';
+  linkType: 'Space';
+  id: string;
+}
+export interface ContentTypeLink {
+  type: 'Link';
+  linkType: 'ContentType';
+  id: string;
+}
+export interface Field {
+  disabled: boolean;
+  id: string;
+  linkType?: string;
+  localized: boolean;
+  name: string;
+  omitted: boolean;
+  required: boolean;
+  type: FieldType;
+  validations: FieldValidation[];
+  items?: FieldItem;
+}
+export declare type FieldType = 'Symbol' | 'Text' | 'Integer' | 'Number' | 'Date' | 'Boolean' | 'Location' | 'Link' | 'Array' | 'Object' | 'RichText';
+export interface FieldValidation {
+  unique?: boolean;
+  size?: {
+    min?: number;
+    max?: number;
+  };
+  regexp?: {
+    pattern: string;
+  };
+  linkMimetypeGroup?: string[];
+  in?: string[];
+  linkContentType?: string[];
+  message?: string;
+  nodes?: {
+    'entry-hyperlink'?: FieldValidation[];
+    'embedded-entry-block'?: FieldValidation[];
+    'embedded-entry-inline'?: FieldValidation[];
+  };
+  enabledNodeTypes?: string[];
+}
+export interface FieldItem {
+  type: 'Link' | 'Symbol';
+  validations: FieldValidation[];
+  linkType?: 'Entry' | 'Asset';
+}
+/**
+ * Types of fields found in an Entry
+ */
+export declare namespace EntryFields {
+  type Symbol = string;
+  type Text = string;
+  type Integer = number;
+  type Number = number;
+  type Date = string;
+  type Boolean = boolean;
+  interface Location {
+    lat: string;
+    lon: string;
+  }
+  type Link<T> = Asset | Entry<T>;
+  type Array<T = any> = Symbol[] | Entry<T>[] | Asset[];
+  type Object<T = any> = T;
+  interface RichText {
+    data: {};
+    content: RichTextContent[];
+    nodeType: 'document';
+  }
+}
+interface RichTextDataTarget {
```

Comment:
Shall we already import the rich text types from the separate project here?

Reply:
@TimBeyer I think it's worth a try, but we should do it in a separate PR